src/extension.ts

```diff
@@ -464,8 +470,16 @@ function getSonarLintConfiguration(): VSCode.WorkspaceConfiguration {
 }
 
 export function deactivate(): Thenable<void> {
-  if (!languageClient) {
-    return undefined;
+  if (languageClient) {
+    return languageClient.stop()
+      .then(disposeOutput);
   }
-  return languageClient.stop();
+  return Promise.resolve(disposeOutput());
+}
+
+function disposeOutput() {
```

Comment:
Why not using `context.subscriptions.push(sonarlintOutput);` in the activate instead?

Reply:
Good catch, I had not seen this in the docs.